gdx/src/com/badlogic/gdx/maps/tiled/BaseTmjMapLoader.java

```diff
@@ -0,0 +1,746 @@
+					tiledMapTileMapObject.setScaleX(scaleX * (objectWidth / textureRegion.getRegionWidth()));
+					tiledMapTileMapObject.setScaleY(scaleY * (objectHeight / textureRegion.getRegionHeight()));
+					tiledMapTileMapObject.setRotation(element.getFloat("rotation", 0));
+					object = tiledMapTileMapObject;
+				} else {
+					object = new RectangleMapObject(x, flipY ? y - height : y, width, height);
+				}
+			}
+			object.setName(element.getString("name", null));
+			String rotation = element.getString("rotation", null);
+			if (rotation != null) {
+				object.getProperties().put("rotation", Float.parseFloat(rotation));
+			}
+			String type = element.getString("type", null);
+			if (type != null) {
+				object.getProperties().put("type", type);
+			}
+			int id = element.getInt("id", 0);
+			if (id != 0) {
+				object.getProperties().put("id", id);
+			}
+			object.getProperties().put("x", x);
+
+			if (object instanceof TiledMapTileMapObject) {
+				object.getProperties().put("y", y);
+			} else {
+				object.getProperties().put("y", (flipY ? y - height : y));
+			}
+			object.getProperties().put("width", width);
+			object.getProperties().put("height", height);
+			object.setVisible(element.getBoolean("visible", true));
+			JsonValue properties = element.get("properties");
+			if (properties != null) {
+				loadProperties(object.getProperties(), properties);
+			}
+			idToObject.put(id, object);
+			objects.add(object);
+		}
+	}
+
+	private void loadProperties (final MapProperties properties, JsonValue element) {
+		if (element == null) return;
+
+		if (element.name() != null && element.name().equals("properties")) {
+			for (JsonValue property : element) {
+				final String name = property.getString("name", null);
+				String value = property.getString("value", null);
+				String type = property.getString("type", null);
+				if (value == null) {
+					value = property.asString();
+				}
+				if (type != null && type.equals("object")) {
+					// Wait until the end of [loadTiledMap] to fetch the object
+					try {
+						// Value should be the id of the object being pointed to
+						final int id = Integer.parseInt(value);
+						// Create [Runnable] to fetch object and add it to props
+						Runnable fetch = new Runnable() {
+							@Override
+							public void run () {
+								MapObject object = idToObject.get(id);
+								properties.put(name, object);
+							}
+						};
+						// [Runnable] should not run until the end of [loadTiledMap]
+						runOnEndOfLoadTiled.add(fetch);
+					} catch (Exception exception) {
+						throw new GdxRuntimeException(
+							"Error parsing property [\" + name + \"] of type \"object\" with value: [" + value + "]", exception);
+					}
+				} else {
+					Object castValue = castProperty(name, value, type);
+					properties.put(name, castValue);
+				}
+			}
+		}
+	}
+
+	private Object castProperty (String name, String value, String type) {
+		if (type == null) {
```

Comment:
The TMX and TMJ files differ once again over here. In the TMJ file it explicitly denotes String properties as type="string" while in the TMX file there is no type whatsoever, which is why it seems it was chosen to compare against null. 
Since this is fundamentally different we should alter this line in order for it to load String properties properly.
**Replace** if (type == null) { with:
`if (type.equals("string")) {`